storage/replica_leader_lease.go

```diff
@@ -128,6 +135,33 @@ func (p *pendingLeaderLeaseRequest) InitOrJoinRequest(
 	return llChan
 }
 
+// InTransferStasis returns true if the replica cannot propose a Raft command with the
+// given timestamp because its LeaderLease is in the stasis period induced by a
+// lease transfer. If the first result is true, a lease indicating the next
+// leader is also returned.
+//
+// It is assumed that the replica owning this pendingLeaderLeaseRequest owns the
+// LeaderLease.
+//
+// replicaID is the ID of the owning replica.
+func (p *pendingLeaderLeaseRequest) InTransferStasis(
+	timestamp roachpb.Timestamp,
+	replicaID roachpb.ReplicaID,
+	leaseTransferStasisDuration time.Duration,
+) (bool, *roachpb.Lease) {
+	if nextLease := p.RequestPending(); nextLease != nil {
```

Comment:
I'm uncomfortable with this. You're relying on in-memory status to preserve correctness, but a quick node restart is possible.